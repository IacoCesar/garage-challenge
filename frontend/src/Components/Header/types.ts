export interface IHeader {
  testID?: string
  isHomeScreen?: boolean
  title?: string
  callHandleBack?: (navigation: any) => void
  callHandleChangeVisualMode?: (themeSelector: string) => void
}

export interface IProps extends IHeader {
  themeSelector: string
}
