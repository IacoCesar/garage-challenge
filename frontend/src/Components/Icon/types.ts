export interface IProps {
  testID?: string
  size: number
  name: string
  color: string
  type?: string
}
