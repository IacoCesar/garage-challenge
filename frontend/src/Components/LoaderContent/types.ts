export interface IProps {
  testID?: string
  size: number
  color: string
}
