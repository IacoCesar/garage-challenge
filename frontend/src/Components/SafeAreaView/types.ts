export interface IProps {
  testID?: string
  children: JSX.Element
}

export interface IThemeProps {
  themeSelector: string
}
