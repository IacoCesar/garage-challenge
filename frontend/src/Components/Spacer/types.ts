export interface IProps {
  testID?: string
  min: number
  max: number
}
