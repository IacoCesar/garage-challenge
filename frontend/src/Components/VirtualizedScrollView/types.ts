export interface IProps {
  testID?: string
  themeSelector: string
  children?: JSX.Element
}
