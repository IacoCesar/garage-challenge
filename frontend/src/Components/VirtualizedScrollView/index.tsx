import { assignTestId } from '@/Helpers/TestId'
import { RootState } from '@/Store'
import React from 'react'
import { useSelector } from 'react-redux'
import S from './styles'

const Component: React.FC<any> = ({
  testID = 'VirtualizedScrollView',
  children
}) => {
  const themeSelector = useSelector((state: RootState) => state.themeData.theme)

  return (
    <S.List
      {...assignTestId('List', `${testID}`)}
      themeSelector={themeSelector}
      data={[]}
      keyExtractor={() => 'key'}
      renderItem={null}
      ListHeaderComponent={<>{children}</>}
    />
  )
}

export default Component
