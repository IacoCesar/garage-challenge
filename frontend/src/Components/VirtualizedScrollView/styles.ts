import styled from 'styled-components/native'
import { IProps } from './types'

const List = styled.FlatList<IProps>`
  flex: 1;
  background-color: ${({ theme, themeSelector }) =>
    theme.standard[themeSelector].background};
`

export default {
  List
}
