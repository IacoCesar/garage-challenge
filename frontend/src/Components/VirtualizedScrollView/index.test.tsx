import React from 'react'
import { IProps } from './types'
import { mockTestId } from '@/Helpers/TestId'
import VirtualizedScrollView from '.'
import { render } from '@/Helpers/TestHelpers'

describe('VirtualizedScrollView', () => {
  let props: IProps = {
    testID: 'VirtualizedScrollView',
    themeSelector: 'default',
    children: <></>
  }

  const loaderElement = mockTestId('List', `${props.testID!}`).testID

  it(`should render ${loaderElement}`, async () => {
    const wrapper = render(<VirtualizedScrollView {...props} />)

    const el_container = wrapper.getByTestId(loaderElement)
    expect(el_container).toBeTruthy()
  })

  it('should render snapshot', async () => {
    const wrapper = render(<VirtualizedScrollView {...props} />)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
