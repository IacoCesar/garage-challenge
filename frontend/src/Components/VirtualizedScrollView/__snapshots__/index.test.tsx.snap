// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VirtualizedScrollView should render snapshot 1`] = `
<RCTScrollView
  ListHeaderComponent={
    <React.Fragment>
      <React.Fragment />
    </React.Fragment>
  }
  accessible={false}
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  testID="List_VirtualizedScrollView"
  themeSelector="default"
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    />
  </View>
</RCTScrollView>
`;
